components/ai_chat/core/browser/engine/engine_consumer_claude_unittest.cc

```diff
@@ -0,0 +1,140 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/ai_chat/core/browser/engine/engine_consumer_claude.h"
+
+#include <memory>
+#include <optional>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "base/functional/callback_helpers.h"
+#include "base/run_loop.h"
+#include "base/strings/string_util.h"
+#include "base/test/bind.h"
+#include "base/test/task_environment.h"
+#include "brave/components/ai_chat/core/browser/engine/engine_consumer.h"
+#include "brave/components/ai_chat/core/browser/engine/mock_remote_completion_client.h"
+#include "brave/components/ai_chat/core/browser/models.h"
+#include "brave/components/ai_chat/core/common/mojom/ai_chat.mojom.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+#include "testing/gmock/include/gmock/gmock.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+using ConversationHistory = std::vector<ai_chat::mojom::ConversationTurn>;
+using ::testing::_;
+
+namespace ai_chat {
+
+class EngineConsumerClaudeUnitTest : public testing::Test {
+ public:
+  EngineConsumerClaudeUnitTest() = default;
+  ~EngineConsumerClaudeUnitTest() override = default;
+
+  void SetUp() override {
+    auto* model = GetModel("chat-claude-instant");
+    ASSERT_TRUE(model);
+    engine_ =
+        std::make_unique<EngineConsumerClaudeRemote>(*model, nullptr, nullptr);
+    engine_->SetAPIForTesting(
+        std::make_unique<MockRemoteCompletionClient>(model->name));
+  }
+
+  void TearDown() override {}
+
+ protected:
+  base::test::TaskEnvironment task_environment_;
+  std::unique_ptr<EngineConsumerClaudeRemote> engine_;
+};
+
+TEST_F(EngineConsumerClaudeUnitTest, TestGenerateAssistantResponse) {
+  ConversationHistory history = {
+      {mojom::CharacterType::HUMAN, mojom::ActionType::SUMMARIZE_SELECTED_TEXT,
+       mojom::ConversationTurnVisibility::VISIBLE,
+       "Which show is this catchphrase from?", "I have spoken."},
+      {mojom::CharacterType::ASSISTANT, mojom::ActionType::RESPONSE,
+       mojom::ConversationTurnVisibility::VISIBLE, "The Mandalorian.",
+       std::nullopt}};
+  auto* mock_remote_completion_client =
+      static_cast<MockRemoteCompletionClient*>(engine_->GetAPIForTesting());
+  std::string prompt_before_time_and_date =
+      "\n\nHuman: Here is the excerpt of the content in <excerpt> "
+      "tags:\n<excerpt>\nI'm groot.\n</excerpt>\n\nA user selects this "
+      "excerpt from the page content.\n\nHere is the text of a web page in "
```

Comment:
This is the part that I think should be located to just before "Here is the user's question" part, and possibly change it to "Here is the user's question about that excerpt"

Reply:
Thank you for pointing out, was trying to get it adjusted during the review, I'll point it out directly or ask @LorenzoMinto  first next time. Asked him in the DM now, will adjust.

Reply:
Agree with @petemill. We should move the EXCERPT bit below the history and above the user question, and use an excerpt-specific question 👍

Reply:
On a second thought, since excerpts might serve as context for subsequent requests and because it might be the only context available (where no page content is added), I think it might be better to have it as an addendum to the system prompt, but after the page content segment. It might still be worth changing the question formulation to reflect whether <excerpt>/<page> are available. Wdyt @petemill?

Another question, could there be more than one excerpt per conversation?

Reply:
It could be possible in a conversation, there’re other previous selected text in the history, but they're not in excerpt tag anymore, only the latest one will be inside tag.